python/sglang/srt/managers/tokenizer_manager.py

```diff
@@ -183,13 +183,25 @@ async def _handle_single_request(
                 obj.top_logprobs_num if not_use_index else obj.top_logprobs_num[index]
             )
         else:
-            if isinstance(obj.text, list):
-                input_text = obj.text[index]
-                rid = obj.rid[index]
+            if obj.text is not None:
+                if isinstance(obj.text, list):
+                    input_text = obj.text[index]
+                    rid = obj.rid[index]
+                else:
+                    input_text = obj.text
+                    rid = obj.rid[0]
+                input_ids = self.tokenizer.encode(input_text)
             else:
-                input_text = obj.text
-                rid = obj.rid[0]
-            input_ids = self.tokenizer.encode(input_text)
+                input_text = None
+                if isinstance(obj.input_ids, list) and isinstance(
+                    obj.input_ids[0], list
+                ):
+                    # when obj["input_ids"] is list[list[int]]
```

Comment:
This comment is great. Could you add more comments like this?
Cover every branch from `if not is_cache_for_prefill:` to all other cases.